### PR TITLE
rgw/rgw_main: fix unnecessary variables defined

### PR DESCRIPTION
**Pull request by yonghengdexin735**
Signed-off-by: zhang.zezhu <zhang.zezhu@zte.com.cn>

### CODE REVIEW

File: src/rgw/rgw_main.cc [hunk 1]
```diff
@@ -78,9 +78,9 @@ void signal_shutdown()
     int val = 0;
     int ret = write(signal_fd[0], (char *)&val, sizeof(val));
     if (ret < 0) {
-      int err = -errno;
+      ret = -errno;
       derr << "ERROR: " << __func__ << ": write() returned "
-           << cpp_strerror(-err) << dendl;
+           << cpp_strerror(ret) << dendl;
```
Comment (theanalyst) [inline src/rgw/rgw_main.cc]: actually there is no need ot change the sign of ret here,  cpp_strerror already has an internal check for this purpose, so just call cpp_strerror(errno);

**Comment by yonghengdexin735** [comment]
@theanalyst  I think here defined ret for return and then defined err for 'errno' is unnecessary. 
First, the 'errno'  assign to ret (ret = errno). socond, remove 'int err = -errno' directly, just call cpp_strerror(errno). here I use the latter.

**Comment by yonghengdexin735** [comment]
@theanalyst  done it

**Comment by theanalyst** [comment]
lgtm

**Pull request merged**